NitroxServer/Communication/Packets/Processors/ModuleAddedProcessor.cs

```diff
@@ -5,27 +5,20 @@
 
 namespace NitroxServer.Communication.Packets.Processors
 {
-    class EquipmentRemoveItemPacketProcessor : AuthenticatedPacketProcessor<EquipmentRemoveItem>
+    class ModuleAddedProcessor : AuthenticatedPacketProcessor<ModuleAdded>
```

Comment:
Was this only modules? or both equipment and modules?

Reply:
That's a mixed bag, my friend.

The server processor of this packet type was doing different things with it depending on the state of the IsPlayerEquipment property. The intent was to update the persistence layer with the current equipment for the player that sent it, or for a submarine/mech that the player was interacting with. From there, things got a little more convoluted.

After the server processed the packet, it forwarded it on to all of the other players. However;  the client processor that received this packet only worked for modules because it did a container lookup with the owner GUID and remote players do not have a container attached to their models. 

Therefore: I believe the intent was that it could work for both player equipment and modules, but the way that Subnautica treats player equipment differently than modules made the intent confusing. 

Given that, I went ahead and separated the concerns so that the intent of each packet was clear from beginning to end. Modules need to be tracked as game objects for each instance of the owner on each client. Player equipment; however, is just a matter of changing the rendered state of game objects that are already attached to every remote player.